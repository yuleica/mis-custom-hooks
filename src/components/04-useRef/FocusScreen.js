import React, { useRef } from 'react';

import '../02-useEffect/effects.css';

export const FocusScreen = () => {

    const inputRef = useRef();

    const handleClick = () => {
        //document.querySelector('input').focus();
        inputRef.current.focus();
    }

    return (
        <div>
            <h1>FocusScreen</h1>
            <hr />

            <input
                ref={inputRef}
                className="form-control"
                placeholder="su nombre" />


            <button 
                className="btn btn-outline-primary mt-3"
                onClick={handleClick}
                >Focus</button>

        </div>
    )
}
