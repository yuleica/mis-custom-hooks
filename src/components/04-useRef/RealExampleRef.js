
import React, {useState} from 'react';
import {MultipleCustomHooks} from '../03-examples/MultipleCustomHooks';
import '../02-useEffect/effects.css';

export const RealExampleRef = () => {
   
   const [show, setShow] = useState(false);
   
    return (
        <div>
            <h1>Real Example Ref</h1>
            <hr />
            {/*si show es true, muestra el componente */}
            { show && <MultipleCustomHooks /> }

            <button
                className="btn btn-primary mt-3"
                onClick={ () => {setShow(!show);}}
            >show/hide</button>

        </div>
    )
}
