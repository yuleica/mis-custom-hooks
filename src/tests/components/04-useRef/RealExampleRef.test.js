import React from 'react';
import { shallow } from 'enzyme';
import {RealExampleRef } from '../../../../src/components/04-useRef/RealExampleRef';


//lo defino aqu√≠ porque no lo voy a cambiar dentro
//del test

const wrapper = shallow(<RealExampleRef />);

test('should show correctly', () => {
    
    expect(wrapper).toMatchSnapshot()
    expect(wrapper.find('MultipleCustomHooks').exists()).toBe(false);
})

test('should show MultipleCustomHooks ', () => {
    
    wrapper.find('button').simulate('click');
    expect(wrapper.find('MultipleCustomHooks').exists()).toBe(true);
})
