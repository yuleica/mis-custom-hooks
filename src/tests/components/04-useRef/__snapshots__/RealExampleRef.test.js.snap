// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should show correctly 1`] = `
<div>
  <h1>
    Real Example Ref
  </h1>
  <hr />
  <button
    className="btn btn-primary mt-3"
    onClick={[Function]}
  >
    show/hide
  </button>
</div>
`;
